### DISCUSSION TITLE
Just taken possession of a couple of 64x64 matrices

### DISCUSSION THREAD

**Discussion by mikejohnau** [PlummersSoftwareLLC/NightDriverStrip#526] [General]
Any idea on where I should start?
/NightDriverStrip/include/MatrixHardware_ESP32_Custom.h. - no clues here
/NightDriverStrip/include/globals.h - nada
Looked at the SmartMatrix include - but do I need to be changing includes?

Guidance would be appreciated.

**Comment by smilima** [comment]
I got it working.  See this thread:

https://github.com/PlummersSoftwareLLC/NightDriverStrip/issues/397

**Reply by mikejohnau** [reply]
Thanks. I'd already set the matrix height and changed the panel type. I just hadnt realised that you could only load a single effect..All working now.

**Comment by smilima** [comment]
Also, if you run into any issues, post here and I will help you out.

**Reply by heidepiek** [reply]
Andrew, can u use the web ui? My 64 by 64 is running mesmerizer, but than the web ui is not reachable. Also tried 128 by 64, than the esp crashes. Tried a couple of times now but esp is not stable. In the serial i see all kinds of things that normally don`t happen. I`m happy with my 64x32 but i was curious if it would run two 64x64 matrix panels.

Harrie

**Comment by mikejohnau** [comment]
Ok. I have the 64x64 matrix working. Most effects seem ok. But the smoke effect is wierd. I have had a look at the code and cant see anything that looks like it may cause this.

https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/26905608/3890110a-ef98-4420-9a42-e43d738cc517

Does anybody have any ideas??

**Comment by robertlipe** [comment]
Smoke is a recent battleground. I tried
<https://github.com/PlummersSoftwareLLC/NightDriverStrip/commit/2b864806cda7d8b1dc064a4d6cda4715e2fd2d10>
making the diagonal lines less harsh and thought I'd succeeded, but they're
back now.


In my investigation in the aobve PR, the black bars are caused by it trying
to iterate through parts of the noise array that are only partially
initialized. FillNoise() and the code that use the noise array have
different ideas of how much of the arrays should be used. Having 2x as many
pixels doesn't help much, does it? I'll admit to not really understanding
the whole noise concept.

For entertainment, try rolling it back to the parent of
https://github.com/PlummersSoftwareLLC/NightDriverStrip/commit/9e1bf8fa428a7945858fe48f60d3aeef0cc5357d
and see if it's better or worse. The noise replacement there may be
suspect.


In general, for the 64x64 panels, you can expect some surprises. Not all
the effects are well exercised in that combination. I'd expect that
direction to be more troublesome than trying to run most of the matrix
displays on a configuration like (a superset of) spectrum with 16x48
WS2812's. Not all of the math is going to Just Work.

RJL

**Comment by davepl** [comment]
It would be great to fix anything that doesn't "just work" on 64x64!  Or at least raise issues...

**Comment by robertlipe** [comment]
Many of the *SM* effects should even work on a Spectrum-like configuration
with 48x16. Obviously, one would be on the hook for hacking up effects.cpp
to hoist the appropriate #includes and ADD_EFFECT (and anything else) out
of the USE_HUB75 and into the custom_effects.h block that's active for the
new configuration that you created (yes, this is a homework assignment for
anyone that wants it - I'm just the idea guy today) to get them to work.

I'd expect the ones that are very "mathy" (MetaBalls, BlurringColors,
FlowFields) to work out but some effects like HolidayLights just won't
handle an aspect ratio like that and ones that need moderate resolution
(StarDeep) would probably look funky on 2812's.

Like so much in this project, I think there's a good opportunity for
tinkering if someone (else) wants to take it. If you want to stack four of
those 16x16 panels for a 32x32 display, that's probably pretty realistic
from a single data pin.

A 3x3 grid is more 2812's than you'd want on a single pin for a reasonable
frame rate. (Here's another homework project: modify class LEDStripGFX to
allow a split of the effects framebuffer across multiple channels so the
effects can dra into a 48x48 grid but you get something like
our ADD_CHANNEL() that splits the calls to FastLED.addLeds() across three
different hardware pins, each with a different offset, length tuple into
the shared frame buffer. That would be ... interesting.

Dave, since it's come up a few times, do you know of a good (open source,
cross platform - oh, heck, at least for MacOS)  viewer that works with the
colorserver on port 49153, err, 12000? I'd like the opposite of
tools/videoserver; I'd often like to "VNC" to an ESP32 that may be running
a configuration of bulbs I don't want to physically attach. Do you have a
viewer?

I once tried writing a thing to connect to the port with the intention of
splashing pixels into Qt or something and the first read was clearly a
ColorDataPacket, but it quickly lost sync and I started getting reads of
crazy sizes and couldn't resync. I got distracted (as I do) and never
followed through.

It'd be handy to document effects, too...

RJL

**Comment by bullz32** [comment]
hi there, i dont have much experience with such things and i cant figure out how to compile it for my panel. can someone kindly provide me a bin file for the esp32 so that i can confirm it it works fine. i have a hub75 panel 64x64 1/32 scan rate. i would appreciate it . also kindly provide some info regarding how to control the panel effects, add new ones etc

**Comment by robertlipe** [comment]
Right hand panel of the landing page on GitHub,

https://plummerssoftwarellc.github.io/NightDriverStrip/

You'll have to compile your own for 64x64.

Instructions are on the landing page at
https://github.com/PlummersSoftwareLLC/NightDriverStrip?tab=readme-ov-file#build-pointers
